File: components/interactive-card/python/ciata_card.py
"""Implementação experimental do CMP-0010 Card interativo para wxPython."""

from __future__ import annotations

from collections.abc import Callable

import wx


class CiataCard(wx.Panel):
    def __init__(
        self,
        parent: wx.Window,
        title: str,
        description: str = "",
        *,
        on_activate: Callable[[], None] | None = None,
    ) -> None:
        super().__init__(parent)

        normalized_title = title.strip()
        normalized_description = description.strip()
        if not normalized_title:
            raise ValueError("title não pode ser vazio.")

        sizer = wx.BoxSizer(wx.VERTICAL)
        self.title = wx.StaticText(self, label=normalized_title)
        sizer.Add(self.title, 0, wx.BOTTOM, 4)
        if normalized_description:
            self.description = wx.StaticText(self, label=normalized_description)
            self.description.Wrap(560)
            sizer.Add(self.description, 0, wx.EXPAND)

        if on_activate is not None:
            self.activate_button = wx.Button(self, label=normalized_title)
            self.activate_button.SetName(normalized_title)
            self.activate_button.Bind(wx.EVT_BUTTON, lambda _event: on_activate())
            sizer.Add(self.activate_button, 0, wx.TOP, 8)

        self.SetSizer(sizer)

File: components/interactive-card/ios/CiataCard.swift
import SwiftUI

/// Implementação experimental do CMP-0010 Card interativo para SwiftUI.
public struct CiataCard: View {
    private let title: String
    private let description: String?
    private let action: (() -> Void)?

    public init(
        title: String,
        description: String? = nil,
        action: (() -> Void)? = nil
    ) {
        let normalizedTitle = title.trimmingCharacters(in: .whitespacesAndNewlines)
        let normalizedDescription = description?.trimmingCharacters(in: .whitespacesAndNewlines)

        precondition(!normalizedTitle.isEmpty, "title não pode ser vazio")
        self.title = normalizedTitle
        self.description = normalizedDescription?.isEmpty == false ? normalizedDescription : nil
        self.action = action
    }

    public var body: some View {
        Group {
            if let action {
                Button(action: action) { content }
                    .buttonStyle(.plain)
            } else {
                content
            }
        }
    }

    private var content: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(title).fontWeight(.semibold)
            if let description {
                Text(description)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding()
    }
}
